src/controllers/midi/midicontroller.cpp

```diff
@@ -288,7 +288,7 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
             return;
         }
 
-        QScriptValue function = pEngine->resolveFunction(mapping.control.item);
+        QScriptValue function = pEngine->wrapFunctionCode(mapping.control.item, 5);
```

Comment:
IMO this would violate abstraction barriers and instead needs some somewhat involved refactoring to gain the performance wins you're talking about. (Script values escaping the ControllerEngine as they did with resolveFunction was already an abstraction violation and storing the values in the MidiController/preset is taking it a step further :) ).

For example -- we need to invalidate the wrapper script values when the filesystem watcher notices the script has changed. If the QScriptValue was stored in the MidiInputMapping (or an m_incomingDataFunctions list) there is no current easy way for the script engine to reach out into the MidiMapping/MidiController to do this. I'm not even sure if it's safe to use a QScriptValue after its QScriptEngine has been deleted (gracefulShutdown destroys the current engine).